src/main/java/com/softserve/kickscooter/kickscootervehicle/management/service/ScooterManagementService.java

```diff
@@ -0,0 +1,70 @@
+package com.softserve.kickscooter.kickscootervehicle.management.service;
+
+
+import com.softserve.kickscooter.kickscootervehicle.management.dao.Scooter;
+import com.softserve.kickscooter.kickscootervehicle.management.dao.ScooterRepository;
+import com.softserve.kickscooter.kickscootervehicle.management.dto.ScooterCreateDto;
+import com.softserve.kickscooter.kickscootervehicle.management.dto.ScooterInfoDto;
+import lombok.AllArgsConstructor;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class ScooterManagementService implements ManagementService {
+
+    private ScooterRepository scooterRepo;
+    private ConversionService convService;
+    //private ScooterStatusService statusService;
+
+    public Scooter createScooter(ScooterCreateDto dto){
+        Scooter scooter = convService.convert(dto, Scooter.class);
+        scooterRepo.save(scooter);
+        //UUID id = scooter.getId();
+        //statusService.newScooter(id);
+        return scooter;
+    }
+
+    public Optional<ScooterInfoDto> getScooterInfo(UUID id){
+        Optional<Scooter> optSc = scooterRepo.findById(id);
+        if(optSc.isPresent()) {
+            Scooter scooter = optSc.get();
+            return Optional.ofNullable(convService.convert(scooter, ScooterInfoDto.class));
+        } else {
+            return Optional.empty();
+        }
+
+    }
+
+    //todo: pagination
+    public Iterable<ScooterInfoDto> getAllScooterInfo(){
+        Iterable<Scooter> scooters = scooterRepo.findAll();
+        ArrayList<ScooterInfoDto> result = new ArrayList<>();
+        for(Scooter sc : scooters) {
+            result.add(convService.convert(sc, ScooterInfoDto.class));
+        }
+        return result;
+    }
+
+    //todo: custom query
+    public Iterable<UUID> getScootersId(){
+        return scooterRepo.getAliveIds();
+    }
+
+
+    public Boolean deleteScooter(UUID id){
```

Comment:
why not:
```
@Transactional
    public Scooter borrowScooter(UUID id) {
        Optional<Scooter> scooter = scooterRepo.findById(id);

        if (scooter.isEmpty()) {
            throw new ScooterNotFoundException();
        }

        var foundScooter = scooter.get();

        if (foundScooter.isAlive()) {
            throw new ScooterIsAlreadyBorrowedException();
        } else {
            foundScooter.setIsAlive(false);
        }

        return scooterRepo.save(foundScooter);
    }
```

Reply:
also, take care of parallel borrows (read about optimistic/pessimistic locking). Actually, to fix this you need to add a new field to the Scooter entity.